### PR TITLE
Convention-based connection string without manage rights

### PR DESCRIPTION
**Pull request by danielmarbach**
Required to introduce acceptance tests for https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/issues/689

This PR introduces a convention-based connection string whose name is derived from the `connection-string-name` parameter from the action. The main connection string is full manage rights while the restricted connection string only provides `Listen` and `Send` rights. This allows testing more advanced scenarios, such as running an endpoint in least privilege mode.

### CODE REVIEW

**Comment by danielmarbach** [comment]
The NPM updates will get rid of the [warnings](https://github.com/Particular/setup-azureservicebus-action/actions/runs/3499588503/jobs/5861315882#step:4:11). Similar to what I have already done in https://github.com/Particular/setup-ravendb-action/pull/12 or https://github.com/Particular/setup-oracle-action/pull/14

File: README.md [hunk 1]
```diff
@@ -12,6 +12,20 @@ This action handles the setup and teardown of an Azure Service Bus namespace for
...
+
+## Development
+
+Open the folder in Visual Studio Code. If you don't already have them, you will be prompted to install remote development extensions. After installing them, and re-opening the folder in a container, do the following:
+
+Run the npm installation
+
+```bash
+npm install
+```
```
Comment (timbussmann) [inline README.md:29]: can't this be done as part of a [postCreate](https://containers.dev/implementors/json_reference/#lifecycle-scripts) callback?

Comment (danielmarbach) [reply README.md:29]: I think I tried it before and for some reason it didn't work, and I removed it again but I can't recall why.

Comment (danielmarbach) [reply README.md]: I think at the time I also added npm update which caused all sorts of troubles (or maybe I did it in the docker container. I added it and it works like a charm

**Comment by danielmarbach** [comment]
Oh it seems reviews are not required. I just wanted to set it to squad after view but it merged it :(

**Comment by danielmarbach** [comment]
I like how you give me the benefit of the doubt :dancing_men:

**Pull request merged**